tools/lou_checkyaml.c

```diff
@@ -365,20 +365,83 @@ read_flags(yaml_parser_t *parser, int *testmode) {
 	yaml_event_delete(&event);
 }
 
+#define XFAIL_NONE 0
+#define XFAIL_FORWARD 1
+#define XFAIL_BACKWARD 2
+#define XFAIL_BOTH 3
+
 static int
-read_xfail(yaml_parser_t *parser) {
+read_xfail_value(yaml_parser_t *parser) {
 	yaml_event_t event;
-	/* assume xfail true if there is an xfail key */
-	int xfail = 1;
+	int xfail = 0;
+
 	if (!yaml_parser_parse(parser, &event) || (event.type != YAML_SCALAR_EVENT))
 		yaml_error(YAML_SCALAR_EVENT, &event);
 	if (!strcmp((const char *)event.data.scalar.value, "false") ||
 			!strcmp((const char *)event.data.scalar.value, "off"))
```

Comment:
@egli Could it be that the `!strcmp` needs to be `strcmp`?

Reply:
Ah, yes. I think I just pushed a fix